File: 905 - Sort Array By Parity.cpp
class Solution {
public:
    vector<int> sortArrayByParity(vector<int>& A) {
        vector<int> B(A);
        int i = 0, j = A.size() - 1;
        for (int a : A) {
            if (a % 2 == 0) {
                B[i++] = a;
            } else {
                B[j--] = a;
            }
        }
        return B;
    }
};

File: 905 - Sort Array By Parity.py
class Solution:
    def sortArrayByParity(self, A: "list[int]") -> "list[int]":
        i, j = 0, len(A) - 1
        while i < j:
            if A[i] % 2 == 1 and A[j] % 2 == 0:
                A[i], A[j] = A[j], A[i]
            if A[i] % 2 == 0:
                i += 1
            if A[j] % 2 == 1:
                j -= 1
        return A


s = Solution()
a = [3,1,2,4]
s.sortArrayByParity(a)